block_iagora.php

```diff
@@ -49,17 +49,34 @@ public function get_content() {
         $this->content = new stdClass();
         $this->content->footer = '';
 
-        $iframeurl = isset($this->config->iframeurl) ? $this->config->iframeurl : '';
+        $copilotendpointurl = isset($this->config->copilotendpointurl) ? $this->config->copilotendpointurl : '';
 
-        if (empty($iframeurl)) {
-            $this->content->text = get_string('noiframeurl', 'block_iagora');
+        if (empty($copilotendpointurl)) {
+            $this->content->text = get_string('nocopilotendpointurl', 'block_iagora');
         } else {
-            $this->content->text = '<iframe src="' . $iframeurl . '" width="100%" height="400px" frameborder="0"></iframe>';
+            $this->content->text = $this->generate_chat_content($copilotendpointurl);
         }
-
+        
         return $this->content;
     }
-
+    
+    /**
+     * Generate the HTML content for the chat block.
+     *
+     * @param string $copilotendpointurl The URL to be used in the iframe.
+     * @return string The generated HTML content for the chat.
+     */
+    private function generate_chat_content($copilotendpointurl) {
+        global $PAGE, $OUTPUT;
```

Comment:
```suggestion
        global $OUTPUT;
```
`$PAGE` seems to be not used.